src/closure-externs/closure-externs.js

```diff
@@ -260,3 +260,12 @@ var moduleArg;
  */
 Navigator.prototype.webkitGetUserMedia = function(
     constraints, successCallback, errorCallback) {};
+
+/**
+ * Global object for custom event target mappings, see:
+ * https://emscripten.org/docs/api_reference/html5.h.html#how-to-use-this-api
+ *
+ * @type {Object}
+ * @suppress {duplicate}
+ */
+var specialHTMLTargets;
```

Comment:
I'm not sure about `specialHTMLTargets` being an extern.   It looks like its just another JS library symbol like any other: https://github.com/emscripten-core/emscripten/blob/3319a313d3b589624d342b650884caaf8cd9ef30/src/library_html5.js#L318-L322

Perhaps you just need to add a dependency on this symbol?   You can do this using EM_JS_DEPS macro, or you can use the `-sDEFAULT_LIBRARY_FUNCS_TO_INCLUDE`

Reply:
For `specialHTMLTartgets` it's not the dependency, it's the type that Closure expects:
```
WARNING - [JSC_TYPE_MISMATCH] restricted index type
found   : string
required: number
  688|  specialHTMLTargets["!canvas"] = Module["canvas"];
                           ^^^^^^^^^
```
The JS is declaring an array, and, as per the docs, this treating it as an object.

I'll take a look at other options, the annotations don't seem to be getting passed on.

Reply:
Since the code is generated, this works to add the annotation:
```
$specialHTMLTargets__docs: '/** @type {Object} */',
```